packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue

```diff
@@ -31,129 +27,175 @@
-        @click="mainButtonClick"
+        v-if="!$props.leftIcon"
+        :disabled="$props.disabled || $props.disableButton"
+        v-bind="{ ...computedMainButtonProps }"
+        v-on="mainButtonListeners"
       >
         <slot name="label">
           {{ label }}
         </slot>
       </va-button>
+
       <va-dropdown
-        :disabled="disabled || disableDropdown"
-        :position="position"
+        :disabled="$props.disabled || $props.disableDropdown"
+        :position="$props.position"
         :offset="$props.offset"
-        v-model="showDropdown"
-        :stateful="stateful"
+        :stateful="$props.stateful"
+        v-model="valueComputed"
       >
         <template #anchor>
           <va-button
-            :size="size"
-            :flat="flat"
-            :outline="outline"
-            :disabled="disabled || disableDropdown"
-            :color="color"
+            :disabled="$props.disabled || $props.disableDropdown"
             :icon="computedIcon"
-            @click="click"
+            v-on="listeners"
           />
         </template>
         <va-dropdown-content>
           <slot />
         </va-dropdown-content>
       </va-dropdown>
+
+      <va-button
+        v-if="$props.leftIcon"
+        :disabled="$props.disabled || $props.disableButton"
+        v-bind="{ ...computedMainButtonProps }"
+        v-on="mainButtonListeners"
+      >
+        <slot name="label">
+          {{ label }}
+        </slot>
+      </va-button>
     </va-button-group>
   </div>
 </template>
 
 <script lang="ts">
-import { Options, Vue, prop, mixins } from 'vue-class-component'
+import { computed, defineComponent, PropType } from 'vue'
 
-import ColorMixin from '../../services/color-config/ColorMixin'
-import { SizeMixin } from '../../mixins/SizeMixin'
+import { useStatefulProps, useStateful } from '../../composables/useStateful'
+import { useEmitProxy } from '../../composables/useEmitProxy'
 
 import VaDropdown, { VaDropdownContent } from '../va-dropdown'
 import VaButton from '../va-button'
 import VaButtonGroup from '../va-button-group'
 
-class ButtonDropdownProps {
-  modelValue = prop<boolean>({ type: Boolean, default: false })
-  color = prop<string>({ type: String, default: 'primary' })
-  outline = prop<boolean>({ type: Boolean, default: false })
-  disableButton = prop<boolean>({ type: Boolean, default: false })
-  disableDropdown = prop<boolean>({ type: Boolean, default: false })
-  flat = prop<boolean>({ type: Boolean, default: false })
-  disabled = prop<boolean>({ type: Boolean, default: false })
-  size = prop<string>({
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
+const { createEmits, createVOnListeners: createListeners } = useEmitProxy(
+  ['click'],
+)
+
+const { createEmits: createMainButtonEmits, createVOnListeners: createMainButtonListeners } = useEmitProxy(
+  [['click', 'main-button-click']],
```

Comment:
I was a bit confused about it. Can we use here object with keys instead?

```suggestion
  [{ listen: 'click', emit: 'main-button-click' }],
```

Reply:
Replaced array with object